core/modules/qdisp/JobQuery.cc

```diff
@@ -136,12 +138,19 @@ bool JobQuery::cancel() {
         std::lock_guard<std::recursive_mutex> lock(_rmutex);
         // If _queryRequestPtr is not nullptr, then this job has been passed to xrootd and
         // cancellation is complicated.
+        bool cancelled = false;
         if (_queryRequestPtr != nullptr) {
             LOGS_DEBUG(getIdStr() << " cancel QueryRequest in progress");
-            _queryRequestPtr->cancel();
-        } else {
+            if (_queryRequestPtr->cancel()) {
+                LOGS_DEBUG(getIdStr() << " cancelled by QueryRequest");
```

Comment:
Why is it `_idStr` in some places and `getIdStr()` in other?
And there s also `LOGS_DEBUG()` vs `LOGS(_log, LOG_LVL_DEBUG, ...)`.